web/src/enterprise/campaigns/detail/CampaignSpecTab.tsx

```diff
@@ -0,0 +1,31 @@
+import FileDownloadIcon from 'mdi-react/FileDownloadIcon'
+import React, { useMemo } from 'react'
+import { CampaignFields } from '../../../graphql-operations'
+
+export interface CampaignSpecTabProps {
+    originalInput: CampaignFields['currentSpec']['originalInput']
+}
+
+export const CampaignSpecTab: React.FunctionComponent<CampaignSpecTabProps> = ({ originalInput }) => {
+    const downloadUrl = useMemo(() => 'data:text/plain;charset=utf-8,' + encodeURIComponent(originalInput), [
+        originalInput,
+    ])
+    return (
+        <>
+            <div className="d-flex justify-content-between align-items-center mb-2 test-campaigns-spec">
+                <p className="m-0">This campaign was created from the folowing spec:</p>
+                <a
+                    download="campaign-spec.yaml"
```

Comment:
Can we interpolate the campaign name here? That's a nice touch, IMHO, otherwise you get into this whole "Overwrite file?" business when downloading and you'll never know which one is which.

```suggestion
                    download=`${campaignName}.spec.yaml`
```